### ISSUE TITLE
Cannot compile native Skia, python3 does not exist.

### ISSUE THREAD

**Issue by BradChase2011** [mono/SkiaSharp#2674]
### Description

We use custom skia mods and have an older version of SkiaSharp working with it for quite some time.  We are now trying to update skiasharp and while skiasharp compiles, we cannot get skia to compile.  It keeps coming down to this error:

```
PS C:\Development\Examples\SkiaSharpNew> dotnet cake --target=externals
The assembly 'Cake.Xamarin, Version=3.1.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (0.33.0).
For best compatibility it should target Cake.Core version 2.0.0.
The assembly 'Cake.XCode, Version=5.0.0.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (1.0.0).
For best compatibility it should target Cake.Core version 2.0.0.
The assembly 'Cake.FileHelpers, Version=4.0.1.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (1.0.0).
For best compatibility it should target Cake.Core version 2.0.0.
The assembly 'Cake.Json, Version=6.0.1.0, Culture=neutral, PublicKeyToken=null'
is referencing an older version of Cake.Core (1.0.0).
For best compatibility it should target Cake.Core version 2.0.0.
Arguments:
    target                         externals
Source Control:
    PREVIEW_LABEL                  preview
    FEATURE_NAME
    BUILD_NUMBER                   0
    GIT_SHA
    GIT_BRANCH_NAME
    GIT_URL

========================================
externals-android
========================================
Arguments:
    target                         Default

========================================
git-sync-deps
========================================
An error occurred when executing task 'git-sync-deps'.
Error: An error occurred trying to start process '"python3"' with working directory 'C:/Development/Examples/SkiaSharpNew/externals/skia'. The system cannot find the file specified.
An error occurred when executing task 'externals-android'.
Error: .NET CLI: Process returned an error (exit code 1).
```

### Code

No code, this is compile time error.

### Expected Behavior

Expected a build.

### Actual Behavior

Build fails.

### Version of SkiaSharp

2.88.3 (Current)

### Last Known Good Version of SkiaSharp

2.88.2 (Previous)

### IDE / Editor

Visual Studio (Windows)

### Platform / Operating System

Windows

### Platform / Operating System Version

Visual Studio 2022 17.8.0, Windows 11 x64

### Devices

PC

### Relevant Screenshots

![image](https://github.com/mono/SkiaSharp/assets/13279060/ef2a67cd-217f-4db3-9042-88e122473340)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by BradChase2011** [comment]
I got a little further, apparently I had to set the environment variable (PYTHON_EXE) for the python exe instead of just the path variable.  I updated dependencies again, and after that I was missing choco and ninja.

I am wondering if the install docs need updated?  They are the same as when we first built our own libraries and they are MUCH different now.  But good news is, I am compiling now, so I guess this can be closed.

**Comment by BradChase2011** [comment]
Compiling after setting the PYTHON_EXE environment variable to the actual exe instead of only the path.

**Comment by BradChase2011** [comment]
I jumped the gun, externals-windows works but android does not, python3 error again:

========================================
libSkiaSharp
========================================
Done. Made 113 targets from 55 files in 234ms
                                             ninja: Entering directory `out/android/x86'
[907/1395] link libskcms.a
FAILED: libskcms.a
cmd.exe /c  python3 "C:/Development/Examples/SkiaSharpNew/externals/skia/gn/rm.py" "libskcms.a" && C:/Development/Android/ndk-bundle/toolchains/llvm/prebuilt/windows-x86_64/bin/llvm-ar.exe rcs libskcms.a @libskcms.a.rsp
'python3' is not recognized as an internal or external command,
operable program or batch file.
[932/1395] compile ../../../modules/skottie/src/effects/SphereEffect.cpp
ninja: build stopped: subcommand failed.
An error occurred when executing task 'libSkiaSharp'.
Error: Process 'C:/Program Files/Python312/Python.exe' failed with error: 1
An error occurred when executing task 'externals-android'.
Error: .NET CLI: Process returned an error (exit code 1).

**Comment by mattleibow** [comment]
Do you have python3 installed into the PATH variable?

**Comment by BradChase2011** [comment]
> Do you have python3 installed into the PATH variable?

Yea sorry its been awhile since I looked at this but we are fully compiling now.  The issue I think that needs resolved is an update to the docs as they appear really outdated.  We saved a list in a "How To" so we can just keep coming back to that as its hard to remember the entire list when you only re compile every so often on updates.  We have a full list of packages and programs to install and then a build list of environment variables to set as well as what not to install (Tizen) and what to skip (Tizen).  Tizen has some install issues right now in VS and windows that basically kills any other development on mobile including android.  They are aware of the issue but arent willing to fix it (we used to restore backups on a build machine after builds where Tizen wasnt installed).  We dont target Tizen anyways, and really just need the info in the build list.

So again, I think its just the build docs need updated with everything needed like LLVM, choco, Python, Ninja, Tizen, Node.js, on and on...  As well as all the environment settings to build each time, for us for some reason we have to tell it where everything is like 'PYTHON_EXE', 'LLVM_HOME', 'ANDROID_NDK_HOME', as well as adding nodejs and Python312 paths.
